app/models/cloud_tenant.rb

```diff
@@ -8,7 +8,6 @@ class CloudTenant < ApplicationRecord
   has_many   :cloud_subnets
   has_many   :network_ports
   has_many   :network_routers
-  has_many   :shared_cloud_networks, -> { where :shared => true }, :through => :ext_management_system, :source => :cloud_networks
```

Comment:
ok, so, how about renaming the `has_many :cloud_networks` association to `has_many :direct_cloud_networks`
then renaming `def all_cloud_networks` to `def cloud_networks`
then you can get rid of all those method aliases, which look strange, because of the `all_` prefix